### DISCUSSION TITLE
TASK ERROR: can't lock file '/var/lock/qemu-server/lock-103.conf' - got timeout

### DISCUSSION THREAD

**Discussion by tomhusband** [tteck/Proxmox#2779] [Help Needed]
I set up a VM using your Ubuntu VM script with the default options on my Beelink SEi12 Mini PC. After it was running I decided I wasn't going to use the VM so I tried to shut it down before removing but couldn't. All I get is the error message:

```
trying to acquire lock...

TASK ERROR: can't lock file '/var/lock/qemu-server/lock-103.conf' - got timeout

```
I have another VM and a few LXC (all tteck) that have been running great for some time now although I haven't tried shutting them down (I'm a little hesitant). I finally got into the server via ssh and ran `systemctl restart pveproxy pvedaemon pvestatd`
It stopped the hang up but I still can't remove the VM. I can't shutdown, pause, stop or reboot.

When I start Proxmox in Syslog I get: `Apr 11 17:44:21 pve pvedaemon[15053]: VM 103 qmp command failed - VM 103 qmp command 'guest-ping' failed - got timeout`

I just want to remove the VM. Here's my Syslog for that time frame when everything locked up:
```
Apr 11 15:56:26 pve pvedaemon[1070]: <root@pam> successful auth for user 'root@pam'

Apr 11 15:56:40 pve pvedaemon[1070]: VM 103 qmp command failed - VM 103 qmp command 'guest-ping' failed - got timeout

Apr 11 15:56:48 pve pvedaemon[1070]: <root@pam> starting task UPID:pve:000032E4:0003AB3B:6617FA30:qmshutdown:103:root@pam:

Apr 11 15:56:48 pve pvedaemon[13028]: shutdown VM 103: UPID:pve:000032E4:0003AB3B:6617FA30:qmshutdown:103:root@pam:

Apr 11 15:56:59 pve pvedaemon[1071]: VM 103 qmp command failed - VM 103 qmp command 'guest-ping' failed - got timeout

Apr 11 15:57:18 pve pvedaemon[1070]: VM 103 qmp command failed - VM 103 qmp command 'guest-ping' failed - got timeout

Apr 11 15:57:37 pve pvedaemon[1071]: VM 103 qmp command failed - VM 103 qmp command 'guest-ping' failed - unable to connect to VM 103 qga socket - timeout after 31 retries

Apr 11 15:57:50 pve pvedaemon[1070]: <root@pam> starting task UPID:pve:000033B0:0003C324:6617FA6E:qmshutdown:103:root@pam:

Apr 11 15:57:50 pve pvedaemon[13232]: shutdown VM 103: UPID:pve:000033B0:0003C324:6617FA6E:qmshutdown:103:root@pam:

Apr 11 15:57:56 pve pvedaemon[1070]: VM 103 qmp command failed - VM 103 qmp command 'guest-ping' failed - unable to connect to VM 103 qga socket - timeout after 31 retries

Apr 11 15:58:00 pve pvedaemon[13232]: can't lock file '/var/lock/qemu-server/lock-103.conf' - got timeout

Apr 11 15:58:00 pve pvedaemon[1070]: <root@pam> end task UPID:pve:000033B0:0003C324:6617FA6E:qmshutdown:103:root@pam: can't lock file '/var/lock/qemu-server/lock-103.conf' - got timeout

Apr 11 15:58:16 pve pvedaemon[1071]: VM 103 qmp command failed - VM 103 qmp command 'guest-ping' failed - unable to connect to VM 103 qga socket - timeout after 31 retries

Apr 11 15:58:36 pve pvedaemon[1070]: VM 103 qmp command failed - VM 103 qmp command 'guest-ping' failed - unable to connect to VM 103 qga socket - timeout after 31 retries
```

Any hints greatly appreciated.

**Comment by tteck** [comment]
It looks like you never setup Cloud-Init or installed the qemu-guest-agent 🤷🏻‍♂️ 
(If the above is true, the commands below may not work)

To unlock the VM

```
qm unlock 103
```
To delete the VM
```
qm destroy 103 --purge
```

**Comment by tomhusband** [comment]
I guess the above is true because the commands didn't work. Is Cloud-init and qemu-guest-agent something beyond the initial Proxmox installation? Is it something I can set up in Proxmox?

Many thanks for the help.

**Reply by tteck** [reply]
You did run the commands in the Proxmox shell, right?

The instructions state More Info at `https://github.com/tteck/Proxmox/discussions/2072`

**Reply by tomhusband** [reply]
Yes.
`Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.
Last login: Fri Apr 12 13:14:36 BST 2024 from 192.168.1.44 on pts/0
root@pve:~# qm unlock 103
root@pve:~# qm destroy 103 --purge
VM 103 is running - destroy failed
root@pve:~# `

**Reply by tteck** [reply]
If the unlock worked, use the UI to stop the VM. Shutdown will not work.

**Reply by tomhusband** [reply]
The unlock worked (I think) but then shutdown in the UI just locks it up again.

**Reply by tteck** [reply]
> use the UI to **stop** the VM. **Shutdown** will not work.

**Reply by tomhusband** [reply]
Sorry, stop worked. The VM is now stopped. Can I use the remove now?

**Reply by tteck** [reply]
yes

**Reply by tomhusband** [reply]
Whew! Finally. Thanks a lot!

I need to read up on Cloud-init and qemu-guest-agent and get it installed I guess. I haven't had problems like this on the other VM and containers .

**Reply by tteck** [reply]
https://github.com/tteck/Proxmox/discussions/2072

**Reply by tomhusband** [reply]
Right. That, I hate to say, doesn't make sense to me but that isn't your problem.

I've appreciated your help.
